File: dotnet/Razorvine.Pickle/Pickle/Pickle/IObjectDeconstructor.cs
/* part of Pickle, by Irmen de Jong (irmen@razorvine.net) */

using System.IO;
// ReSharper disable InconsistentNaming

namespace Razorvine.Pickle {

	/// <summary>
	/// Interface for deconstructed objects used by the pickler, to pickle custom classes. 
	/// </summary>
	public interface IObjectDeconstructor {
		/// <summary>
		/// Get the module of the class being pickled
		/// </summary>
		string get_module();
        /// <summary>
        /// Get the name of the class being pickled
        /// </summary>
        string get_name();
        /// <summary>
        /// Get the deconstructed values, which will be used as arguments for reconstructing
        /// </summary>
        object[] deconstruct(object obj);
	}

}

File: java/src/main/java/net/razorvine/pickle/IObjectDeconstructor.java
package net.razorvine.pickle;

/**
 * Interface for Object Deconstructors that are used by the pickler
 * to create instances of non-primitive or custom classes.
 *
 * @author Irmen de Jong (irmen@razorvine.net)
 */
public interface IObjectDeconstructor {
	/**
	 * Get the module of the class being pickled
	 */
	String getModule();
	/**
	 * Get the name of the class being pickled
	 */
	String getName();
	/**
	 * Deconstructs the arugment of an object. The given args will be used as parameters for the constructor during unpickling.
	 */
	Object[] deconstruct(Object obj) throws PickleException;
	
}
